source/solvers/navier_stokes_base.cc

```diff
@@ -1085,23 +951,21 @@ NavierStokesBase<dim, VectorType, DofsType>::read_checkpoint()
   setup_dofs();
   std::vector<VectorType *> x_system(4);
 
-  auto &     newton_update = this->get_newton_update();
-  VectorType distributed_system(newton_update);
-  VectorType distributed_system_m1(newton_update);
-  VectorType distributed_system_m2(newton_update);
-  VectorType distributed_system_m3(newton_update);
-  x_system[0] = &(distributed_system);
-  x_system[1] = &(distributed_system_m1);
-  x_system[2] = &(distributed_system_m2);
-  x_system[3] = &(distributed_system_m3);
+  // Initialize vectors and store address in x_system prior deserialization
+  VectorType present_solution(locally_owned_dofs,
+                              locally_relevant_dofs,
+                              mpi_communicator);
+  this->solution_m1.reinit(present_solution);
+  this->solution_m2.reinit(present_solution);
+  this->solution_m3.reinit(present_solution);
+
+  x_system[0] = &(present_solution);
+  x_system[1] = &(this->solution_m1);
+  x_system[2] = &(this->solution_m2);
+  x_system[3] = &(this->solution_m3);
   parallel::distributed::SolutionTransfer<dim, VectorType> system_trans_vectors(
     this->dof_handler);
   system_trans_vectors.deserialize(x_system);
-  auto &present_solution = this->get_present_solution();
-  present_solution       = distributed_system;
-  this->solution_m1      = distributed_system_m1;
-  this->solution_m2      = distributed_system_m2;
-  this->solution_m3      = distributed_system_m3;
```

Comment:
Like we discussed it is a locally relevant vs locally owned thing, but it could be better documented. I will add some comments to make it clearer :).

Reply:
I changed this so that they are initialized with locally_owned_dofs instead of just being "copy initialized". I think this will also be fast (but mostly cleaner and easier to read).